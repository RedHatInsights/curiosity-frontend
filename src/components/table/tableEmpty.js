import React from 'react';
import PropTypes from 'prop-types';
import { EmptyState, EmptyStateIcon, EmptyStateBody, EmptyStateVariant, Title } from '@patternfly/react-core';
import { EmptyTable as PlatformEmptyTableWrapper } from '@redhat-cloud-services/frontend-components/components/cjs/EmptyTable';

/**
 * Render an empty table.
 *
 * @param {object} props
 * @param {Node|Function} props.icon
 * @param {Node} props.message
 * @param {string} props.tableHeading
 * @param {Node} props.title
 * @param {string} props.variant
 * @returns {Node}
 */
const TableEmpty = ({ icon, message, tableHeading, title, variant }) => (
  <PlatformEmptyTableWrapper>
    <EmptyState variant={variant}>
      {icon && <EmptyStateIcon icon={icon} />}
      <Title headingLevel={tableHeading} size="lg">
        {title}
      </Title>
      <EmptyStateBody>{message}</EmptyStateBody>
    </EmptyState>
  </PlatformEmptyTableWrapper>
);

/**
 * Prop types.
 *
 * @type {{icon: Node|Function, variant: string, message: Node, title: Node, tableHeading: string}}
 */
TableEmpty.propTypes = {
  icon: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
  message: PropTypes.node.isRequired,
  tableHeading: PropTypes.string,
  title: PropTypes.node.isRequired,
  variant: PropTypes.oneOf(Object.keys(EmptyStateVariant))
};

/**
 * Default props.
 *
 * @type {{icon: null, variant: EmptyStateVariant.small, tableHeading: string}}
 */
TableEmpty.defaultProps = {
  icon: null,
  tableHeading: 'h2',
  variant: EmptyStateVariant.small
};

export { TableEmpty as default, TableEmpty };
