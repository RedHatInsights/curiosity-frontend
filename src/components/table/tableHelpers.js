import React from 'react';
import { SortByDirection } from '@patternfly/react-table';

/**
 * Table sorting directions.
 *
 * @type {{asc: SortByDirection.asc, desc: SortByDirection.desc}}
 */
const SortByDirectionVariant = { ...SortByDirection };

/**
 * Table header copy wrapping modifier.
 *
 * @type {{noWrap: string, truncate: string, fitContent: string, breakWord: string, wrap: string}}
 */
const WrapModifierVariant = {
  wrap: 'wrap',
  noWrap: 'nowrap',
  truncate: 'truncate',
  breakWord: 'breakWord',
  fitContent: 'fitContent'
};

/**
 * Allow additional content to display in cells.
 *
 * @param {React.ReactNode|Function|object|*} content
 * @returns {*|string}
 */
const parseContent = content =>
  (React.isValidElement(content) && content) ||
  (typeof content === 'function' && content()) ||
  (typeof content === 'object' && `${content}`) ||
  content ||
  '';

/**
 * Parse table header settings, props.
 *
 * @param {object} params
 * @param {Array} params.columnHeaders
 * @param {boolean} params.isAllSelected
 * @param {boolean} params.isRowExpand
 * @param {Array} params.parsedRows
 * @param {Function} params.onSelect
 * @param {Function} params.onSort
 * @returns {{headerRow: any[], headerSelectProps: {}}}
 */
const tableHeader = ({
  columnHeaders = [],
  isAllSelected = false,
  isRowExpand,
  parsedRows = [],
  onSelect,
  onSort
} = {}) => {
  const updatedColumnHeaders = [];
  const updatedHeaderSelectProps = {};
  const isSelectTable = typeof onSelect === 'function';

  if (isSelectTable) {
    const parsedRowData = parsedRows.map(({ data }) => data || {});
    updatedHeaderSelectProps.onSelect = (_event, isSelected) =>
      onSelect({ data: parsedRowData, isSelected, rowIndex: -1, type: 'all' });
    updatedHeaderSelectProps.isSelected = isAllSelected;
  }

  columnHeaders.forEach((columnHeader, index) => {
    const key = `${window.btoa(columnHeader)}-${index}`;

    if (columnHeader?.content !== undefined) {
      const {
        isSort,
        isSortActive,
        sortDirection = SortByDirection.asc,
        content,
        dataLabel,
        info,
        modifier,
        tooltip,
        ...headerCellData
      } = columnHeader;
      const tempColumnHeader = {
        key,
        content: parseContent(content),
        props: {
          dataLabel,
          info,
          modifier,
          tooltip
        },
        data: headerCellData
      };

      if (typeof onSort === 'function' && (isSort === true || isSortActive === true)) {
        let updatedColumnIndex = index;

        if (isRowExpand) {
          updatedColumnIndex += 1;
        }

        if (isSelectTable) {
          updatedColumnIndex += 1;
        }

        tempColumnHeader.props.sort = {
          columnIndex: updatedColumnIndex,
          sortBy: {},
          onSort: (_event, _colIndex, direction) =>
            onSort({ cellIndex: updatedColumnIndex, data: headerCellData, direction, originalIndex: index })
        };

        if (isSortActive) {
          tempColumnHeader.props.sort.sortBy.index = updatedColumnIndex;
        }

        tempColumnHeader.props.sort.sortBy.direction = sortDirection;
      }

      updatedColumnHeaders.push(tempColumnHeader);
    } else {
      updatedColumnHeaders.push({
        key,
        content: parseContent(columnHeader)
      });
    }
  });

  return {
    headerRow: updatedColumnHeaders,
    headerSelectProps: updatedHeaderSelectProps
  };
};

/**
 * Parse table body settings, props.
 *
 * @param {object} params
 * @param {Function} params.onExpand
 * @param {Function} params.onSelect
 * @param {Array} params.rows
 * @returns {{isExpandableCell: boolean, isSelectTable: boolean, isExpandableRow: boolean, isAllSelected: boolean, rows: any[]}}
 */
const tableRows = ({ onExpand, onSelect, rows = [] } = {}) => {
  const updatedRows = [];
  const isSelectTable = typeof onSelect === 'function';
  let isExpandableRow = false;
  let isExpandableCell = false;
  let selectedRows = 0;

  rows.forEach(({ cells, isDisabled = false, isExpanded = false, isSelected = false, expandedContent, ...rowData }) => {
    const rowObj = {
      key: undefined,
      cells: [],
      select: undefined,
      expand: undefined,
      expandedContent,
      data: rowData
    };
    updatedRows.push(rowObj);
    rowObj.rowIndex = updatedRows.length - 1;
    rowObj.key = `${window.btoa(rowObj)}-${rowObj.rowIndex}`;

    if (isSelectTable) {
      const updatedIsSelected = isSelected ?? false;

      if (updatedIsSelected === true) {
        selectedRows += 1;
      }

      rowObj.select = {
        cells,
        rowIndex: rowObj.rowIndex,
        onSelect: (_event, isRowSelected) =>
          onSelect({ data: rowObj.data, isSelected: isRowSelected, rowIndex: rowObj.rowIndex, type: 'row' }),
        isSelected: updatedIsSelected,
        disable: isDisabled || false
      };
    }

    if (expandedContent && typeof onExpand === 'function') {
      isExpandableRow = true;

      rowObj.expand = {
        rowIndex: rowObj.rowIndex,
        isExpanded,
        onToggle: (_event, rowIndex, isRowToggleExpanded) =>
          onExpand({
            data: rowObj.data,
            isExpanded: isRowToggleExpanded,
            rowIndex: rowObj.rowIndex,
            type: 'row'
          })
      };
    }

    cells?.forEach((cell, cellIndex) => {
      const cellKey = `${window.btoa(cell)}-${rowObj.rowIndex}-${cellIndex}`;
      if (cell?.content !== undefined) {
        const { className, content, dataLabel, isActionCell, noPadding, width, style, ...remainingProps } = cell;
        const cellProps = { className: className || '', dataLabel, isActionCell, noPadding, style: style || {} };
        let updatedWidthClassName;

        // FixMe: PF doesn't appear to apply cell width classNames when less than 10, or consistently
        if (typeof width === 'number') {
          updatedWidthClassName = `pf-m-width-${width}`;
        }

        if (typeof width === 'string' || style) {
          cellProps.style = { ...cellProps.style, width };
        } else if (updatedWidthClassName) {
          cellProps.className = `${cellProps.className || ''} ${updatedWidthClassName}`;
        }

        if (!isExpandableRow && cell?.expandedContent && typeof onExpand === 'function') {
          isExpandableCell = true;
          const updateIsExpanded = cell?.isExpanded ?? false;

          cellProps.compoundExpand = {
            isExpanded: updateIsExpanded,
            onToggle: (_event, rowIndex, isRowToggleExpanded, isCellToggleExpanded) =>
              onExpand({
                cellIndex,
                data: rowObj.data,
                isExpanded: !isCellToggleExpanded,
                rowIndex: rowObj.rowIndex,
                type: 'compound'
              })
          };
        }

        rowObj.cells.push({ ...remainingProps, content: parseContent(content), key: cellKey, props: cellProps });
      } else {
        rowObj.cells.push({
          key: cellKey,
          content: parseContent(cell)
        });
      }
    });
  });

  return {
    isAllSelected: selectedRows === rows.length,
    isExpandableRow,
    isExpandableCell,
    isSelectTable,
    rows: updatedRows
  };
};

const tableHelpers = {
  parseContent,
  tableHeader,
  tableRows,
  SortByDirectionVariant,
  WrapModifierVariant
};

export {
  tableHelpers as default,
  tableHelpers,
  parseContent,
  tableHeader,
  tableRows,
  SortByDirectionVariant,
  WrapModifierVariant
};
