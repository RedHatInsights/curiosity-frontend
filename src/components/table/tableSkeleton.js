import React from 'react';
import PropTypes from 'prop-types';
import { TableVariant } from '@patternfly/react-table';
import { Skeleton, SkeletonSize } from '@redhat-cloud-services/frontend-components/components/cjs/Skeleton';
import Table from './table';
import { translate } from '../i18n/i18n';

/**
 * Render a table with skeleton loaders.
 *
 * @param {object} props
 * @param {string} props.className
 * @param {boolean} props.borders
 * @param {number} props.colCount
 * @param {boolean} props.isHeader
 * @param {number} props.rowCount
 * @param {Function} props.t
 * @param {string} props.variant
 * @returns {Node}
 */
const TableSkeleton = ({ className, borders, colCount, isHeader, rowCount, t, variant }) => {
  const updatedColumnHeaders = [...new Array(colCount)].map(() => <Skeleton size={SkeletonSize.md} />);

  const updatedRows = [...new Array(rowCount)].map(() => ({
    cells: [...new Array(colCount)].map(() => ({ cell: <Skeleton size={SkeletonSize.md} /> }))
  }));

  return (
    <Table
      ariaLabel={t('curiosity-inventory.tableSkeletonAriaLabel')}
      borders={borders}
      className={`curiosity-skeleton-table ${className || ''}`}
      columnHeaders={updatedColumnHeaders}
      isHeader={isHeader}
      rows={updatedRows}
      variant={variant}
    />
  );
};

/**
 * Prop types.
 *
 * @type {{borders: boolean, isHeader: boolean, colCount: number, variant: string, className: string, rowCount: number}}
 */
TableSkeleton.propTypes = {
  borders: PropTypes.bool,
  className: PropTypes.string,
  colCount: PropTypes.number,
  isHeader: PropTypes.bool,
  rowCount: PropTypes.number,
  t: PropTypes.func,
  variant: PropTypes.oneOf([...Object.values(TableVariant)])
};

/**
 * Default props.
 *
 * @type {{t: translate, borders: boolean, isHeader: boolean, colCount: number, variant: null, className: null,
 *     rowCount: number}}
 */
TableSkeleton.defaultProps = {
  borders: true,
  className: null,
  colCount: 1,
  isHeader: true,
  rowCount: 5,
  t: translate,
  variant: null
};

export { TableSkeleton as default, TableSkeleton };
