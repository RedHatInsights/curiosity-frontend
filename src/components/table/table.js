import React from 'react';
import PropTypes from 'prop-types';
import { Grid, GridItem } from '@patternfly/react-core';
import SearchIcon from '@patternfly/react-icons/dist/js/icons/search-icon';
import { Table as PfTable, TableBody, TableHeader, TableVariant } from '@patternfly/react-table';
import _isEqual from 'lodash/isEqual';
import { TableEmpty } from './tableEmpty';
import { helpers } from '../../common';
import { translate } from '../i18n/i18n';

/**
 * A table.
 *
 * @augments React.Component
 * @fires onCollapse
 */
class Table extends React.Component {
  state = {
    updatedColumnHeaders: null,
    updatedRows: null
  };

  componentDidMount() {
    const { updatedRows } = this.state;

    if (updatedRows === null) {
      this.setRowData();
    }
  }

  componentDidUpdate(prevProps) {
    const { columnHeaders, rows } = this.props;

    if (!_isEqual(prevProps.rows, rows) || !_isEqual(prevProps.columnHeaders, columnHeaders)) {
      this.setRowData();
    }
  }

  /**
   * Apply expanded row content.
   *
   * @event onCollapse
   * @param {object} params
   * @param {number} params.index
   * @param {boolean} params.isOpen
   */
  onCollapse = ({ index, isOpen }) => {
    const { updatedRows } = this.state;
    updatedRows[index].isOpen = isOpen;

    if (isOpen) {
      updatedRows[index + 1].fullWidth = false;
      updatedRows[index + 1].cells = [{ title: updatedRows[index + 1].expandedContent }];
    } else {
      updatedRows[index + 1].cells = [{ title: '' }];
    }

    this.setState({
      updatedRows
    });
  };

  /**
   * Convert row objects into the required PF Table format.
   */
  setRowData() {
    const { columnHeaders, rows } = this.props;
    const updatedColumnHeaders = [];
    const updatedRows = [];

    columnHeaders.forEach(columnHeader => {
      updatedColumnHeaders.push(columnHeader);
    });

    rows.forEach(({ expandedContent, cells, isExpanded }) => {
      const rowObj = {
        cells: []
      };
      updatedRows.push(rowObj);

      if (expandedContent) {
        rowObj.isOpen = isExpanded || false;

        updatedRows.push({
          parent: updatedRows.length - 1,
          fullWidth: true,
          cells: [{ title: '' }],
          expandedContent
        });
      }

      cells.forEach(cell => {
        if (cell?.cell) {
          const { cell: contentCell, ...settings } = cell;
          rowObj.cells.push({ title: contentCell, ...settings });
        } else {
          rowObj.cells.push({ title: cell });
        }
      });
    });

    this.setState({
      updatedColumnHeaders,
      updatedRows
    });
  }

  /**
   * Apply props to table.
   *
   * @returns {Node}
   */
  renderTable() {
    const { updatedColumnHeaders, updatedRows } = this.state;
    const { ariaLabel, borders, children, className, isHeader, summary, t, variant } = this.props;
    let emptyTable = null;

    if (!updatedRows?.length) {
      emptyTable = children || (
        <TableEmpty
          icon={SearchIcon}
          title={t('curiosity-inventory.tableEmptyInventoryTitle')}
          message={t('curiosity-inventory.tableEmptyInventoryMessage')}
        />
      );
    }

    return (
      <React.Fragment>
        <PfTable
          className={`curiosity-table${(!borders && '-no-border') || ''} ${className || ''}`}
          borders={borders}
          aria-label={ariaLabel || t('curiosity-inventory.tableAriaLabel', { appName: helpers.UI_DISPLAY_NAME })}
          summary={summary || t('curiosity-inventory.tableSummary')}
          variant={variant}
          onCollapse={(event, index, isOpen, data) => this.onCollapse({ event, index, isOpen, data })}
          rows={(updatedRows?.length && updatedRows) || []}
          cells={updatedColumnHeaders || []}
        >
          {isHeader && <TableHeader />}
          <TableBody />
        </PfTable>
        {emptyTable}
      </React.Fragment>
    );
  }

  /**
   * Render a table.
   *
   * @returns {Node}
   */
  render() {
    return (
      <Grid guttter="sm" className="ins-inventory-list">
        <GridItem span={12}>{this.renderTable()}</GridItem>
      </Grid>
    );
  }
}

/**
 * Prop types.
 *
 * @type {{summary: string, columnHeaders: Array, t: Function, borders: boolean, children: Node,
 *     isHeader: boolean, variant: string, className: string, rows, ariaLabel: string}}
 */
Table.propTypes = {
  ariaLabel: PropTypes.string,
  borders: PropTypes.bool,
  children: PropTypes.node,
  className: PropTypes.string,
  columnHeaders: PropTypes.arrayOf(
    PropTypes.oneOfType([
      PropTypes.node,
      PropTypes.shape({
        title: PropTypes.node
      })
    ])
  ).isRequired,
  isHeader: PropTypes.bool,
  rows: PropTypes.arrayOf(
    PropTypes.shape({
      expandedContent: PropTypes.node,
      cells: PropTypes.arrayOf(
        PropTypes.oneOfType([
          PropTypes.node,
          PropTypes.shape({
            cell: PropTypes.node
          })
        ])
      ),
      isExpanded: PropTypes.bool
    })
  ),
  summary: PropTypes.string,
  t: PropTypes.func,
  variant: PropTypes.oneOf([...Object.values(TableVariant)])
};

/**
 * Default props.
 *
 * @type {{summary: null, t: translate, borders: boolean, children: null, isHeader: boolean,
 *     variant: null, className: null, rows: Array, ariaLabel: null}}
 */
Table.defaultProps = {
  ariaLabel: null,
  borders: true,
  children: null,
  className: null,
  isHeader: true,
  rows: [],
  summary: null,
  t: translate,
  variant: null
};

export { Table as default, Table };
