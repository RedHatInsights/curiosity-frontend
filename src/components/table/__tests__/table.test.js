import React from 'react';
import { shallow } from 'enzyme';
import { TableVariant, SortByDirection } from '@patternfly/react-table';
import { Table } from '../table';

describe('Table Component', () => {
  it('should render a non-connected component', () => {
    const props = {
      columnHeaders: ['lorem', 'ipsum', 'dolor', 'sit']
    };

    const component = shallow(<Table {...props} />);
    expect(component).toMatchSnapshot('non-connected');
  });

  it('should allow variations in table layout', () => {
    const props = {
      columnHeaders: ['lorem ipsum'],
      rows: [{ cells: ['dolor'] }, { cells: ['sit'] }]
    };

    const component = shallow(<Table {...props} />);
    expect(component).toMatchSnapshot('generated rows');

    component.setProps({
      borders: false,
      isHeader: false
    });
    expect(component).toMatchSnapshot('borders and table header removed');

    component.setProps({
      ariaLabel: 'lorem ipsum aria-label',
      summary: 'lorem ipsum summary'
    });
    expect(component).toMatchSnapshot('ariaLabel and summary');

    component.setProps({
      className: 'lorem-ipsum-class',
      variant: TableVariant.compact
    });
    expect(component).toMatchSnapshot('className and variant');
  });

  it('should allow expandable content', () => {
    const props = {
      columnHeaders: ['lorem ipsum'],
      rows: [{ cells: ['dolor'], expandedContent: 'dolor sit expandable content' }, { cells: ['sit'] }]
    };

    const component = shallow(<Table {...props} />);
    expect(component).toMatchSnapshot('expandable content');

    const componentInstance = component.instance();
    componentInstance.onCollapse({ index: 0, isOpen: true });
    expect(component).toMatchSnapshot('expanded row');
  });

  it('should allow sortable content', () => {
    const props = {
      columnHeaders: [{ title: 'lorem ipsum', onSort: jest.fn() }],
      rows: [{ cells: ['dolor'] }, { cells: ['sit'] }]
    };

    const component = shallow(<Table {...props} />);
    expect(component).toMatchSnapshot('sortable content');

    const componentInstance = component.instance();
    componentInstance.onSort({ index: 0, direction: SortByDirection.asc });
    expect(component).toMatchSnapshot('sorted column callback');
    expect(props.columnHeaders[0].onSort).toHaveBeenCalledTimes(1);
  });

  it('should pass child components, nodes when there are no rows', () => {
    const props = {
      columnHeaders: ['lorem ipsum'],
      rows: []
    };

    const component = shallow(<Table {...props}>Loading...</Table>);
    expect(component).toMatchSnapshot('children');
  });
});
