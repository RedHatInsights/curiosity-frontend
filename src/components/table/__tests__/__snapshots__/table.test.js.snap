// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component should allow expandable content: expandable content 1`] = `
<Grid
  className="ins-inventory-list"
>
  <GridItem
    span={12}
  >
    <Table
      aria-label="t(curiosity-inventory.tableAriaLabel, {\\"appName\\":\\"Subscriptions\\"})"
      borders={true}
      canCollapseAll={false}
      canSelectAll={true}
      canSortFavorites={true}
      cells={
        Array [
          Object {
            "title": "lorem ipsum",
          },
        ]
      }
      className="curiosity-table "
      collapseAllAriaLabel=""
      contentId="expanded-content"
      dropdownDirection="down"
      dropdownPosition="right"
      expandId="expandable-toggle"
      gridBreakPoint="grid-md"
      isHeaderSelectDisabled={false}
      isNested={false}
      isStickyHeader={false}
      isTreeTable={false}
      onCollapse={[Function]}
      ouiaSafe={true}
      role="grid"
      rowLabeledBy="simple-node"
      rows={
        Array [
          Object {
            "cells": Array [
              Object {
                "title": "dolor",
              },
            ],
            "isOpen": false,
          },
          Object {
            "cells": Array [
              Object {
                "title": "",
              },
            ],
            "expandedContent": "dolor sit expandable content",
            "fullWidth": true,
            "parent": 0,
          },
          Object {
            "cells": Array [
              Object {
                "title": "sit",
              },
            ],
          },
        ]
      }
      selectVariant="checkbox"
      summary="t(curiosity-inventory.tableSummary)"
      variant={null}
    >
      <TableHeader />
      <TableBody />
    </Table>
  </GridItem>
</Grid>
`;

exports[`Table Component should allow expandable content: expanded row 1`] = `
<Grid
  className="ins-inventory-list"
>
  <GridItem
    span={12}
  >
    <Table
      aria-label="t(curiosity-inventory.tableAriaLabel, {\\"appName\\":\\"Subscriptions\\"})"
      borders={true}
      canCollapseAll={false}
      canSelectAll={true}
      canSortFavorites={true}
      cells={
        Array [
          Object {
            "title": "lorem ipsum",
          },
        ]
      }
      className="curiosity-table "
      collapseAllAriaLabel=""
      contentId="expanded-content"
      dropdownDirection="down"
      dropdownPosition="right"
      expandId="expandable-toggle"
      gridBreakPoint="grid-md"
      isHeaderSelectDisabled={false}
      isNested={false}
      isStickyHeader={false}
      isTreeTable={false}
      onCollapse={[Function]}
      ouiaSafe={true}
      role="grid"
      rowLabeledBy="simple-node"
      rows={
        Array [
          Object {
            "cells": Array [
              Object {
                "title": "dolor",
              },
            ],
            "isOpen": true,
          },
          Object {
            "cells": Array [
              Object {
                "title": "dolor sit expandable content",
              },
            ],
            "expandedContent": "dolor sit expandable content",
            "fullWidth": false,
            "parent": 0,
          },
          Object {
            "cells": Array [
              Object {
                "title": "sit",
              },
            ],
          },
        ]
      }
      selectVariant="checkbox"
      summary="t(curiosity-inventory.tableSummary)"
      variant={null}
    >
      <TableHeader />
      <TableBody />
    </Table>
  </GridItem>
</Grid>
`;

exports[`Table Component should allow sortable content: sortable content 1`] = `
<Grid
  className="ins-inventory-list"
>
  <GridItem
    span={12}
  >
    <Table
      aria-label="t(curiosity-inventory.tableAriaLabel, {\\"appName\\":\\"Subscriptions\\"})"
      borders={true}
      canCollapseAll={false}
      canSelectAll={true}
      canSortFavorites={true}
      cells={
        Array [
          Object {
            "onSort": [MockFunction],
            "title": "lorem ipsum",
            "transforms": Array [
              [Function],
            ],
          },
        ]
      }
      className="curiosity-table "
      collapseAllAriaLabel=""
      contentId="expanded-content"
      dropdownDirection="down"
      dropdownPosition="right"
      expandId="expandable-toggle"
      gridBreakPoint="grid-md"
      isHeaderSelectDisabled={false}
      isNested={false}
      isStickyHeader={false}
      isTreeTable={false}
      onSort={[Function]}
      ouiaSafe={true}
      role="grid"
      rowLabeledBy="simple-node"
      rows={
        Array [
          Object {
            "cells": Array [
              Object {
                "title": "dolor",
              },
            ],
          },
          Object {
            "cells": Array [
              Object {
                "title": "sit",
              },
            ],
          },
        ]
      }
      selectVariant="checkbox"
      sortBy={Object {}}
      summary="t(curiosity-inventory.tableSummary)"
      variant={null}
    >
      <TableHeader />
      <TableBody />
    </Table>
  </GridItem>
</Grid>
`;

exports[`Table Component should allow sortable content: sorted column callback 1`] = `
<Grid
  className="ins-inventory-list"
>
  <GridItem
    span={12}
  >
    <Table
      aria-label="t(curiosity-inventory.tableAriaLabel, {\\"appName\\":\\"Subscriptions\\"})"
      borders={true}
      canCollapseAll={false}
      canSelectAll={true}
      canSortFavorites={true}
      cells={
        Array [
          Object {
            "onSort": [MockFunction] {
              "calls": Array [
                Array [
                  Object {
                    "direction": "asc",
                    "index": 0,
                    "label": undefined,
                  },
                ],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            },
            "title": "lorem ipsum",
            "transforms": Array [
              [Function],
            ],
          },
        ]
      }
      className="curiosity-table "
      collapseAllAriaLabel=""
      contentId="expanded-content"
      dropdownDirection="down"
      dropdownPosition="right"
      expandId="expandable-toggle"
      gridBreakPoint="grid-md"
      isHeaderSelectDisabled={false}
      isNested={false}
      isStickyHeader={false}
      isTreeTable={false}
      onSort={[Function]}
      ouiaSafe={true}
      role="grid"
      rowLabeledBy="simple-node"
      rows={
        Array [
          Object {
            "cells": Array [
              Object {
                "title": "dolor",
              },
            ],
          },
          Object {
            "cells": Array [
              Object {
                "title": "sit",
              },
            ],
          },
        ]
      }
      selectVariant="checkbox"
      sortBy={
        Object {
          "direction": "asc",
          "index": 0,
        }
      }
      summary="t(curiosity-inventory.tableSummary)"
      variant={null}
    >
      <TableHeader />
      <TableBody />
    </Table>
  </GridItem>
</Grid>
`;

exports[`Table Component should allow variations in table layout: ariaLabel and summary 1`] = `
<Grid
  className="ins-inventory-list"
>
  <GridItem
    span={12}
  >
    <Table
      aria-label="lorem ipsum aria-label"
      borders={false}
      canCollapseAll={false}
      canSelectAll={true}
      canSortFavorites={true}
      cells={
        Array [
          Object {
            "title": "lorem ipsum",
          },
        ]
      }
      className="curiosity-table-no-border "
      collapseAllAriaLabel=""
      contentId="expanded-content"
      dropdownDirection="down"
      dropdownPosition="right"
      expandId="expandable-toggle"
      gridBreakPoint="grid-md"
      isHeaderSelectDisabled={false}
      isNested={false}
      isStickyHeader={false}
      isTreeTable={false}
      ouiaSafe={true}
      role="grid"
      rowLabeledBy="simple-node"
      rows={
        Array [
          Object {
            "cells": Array [
              Object {
                "title": "dolor",
              },
            ],
          },
          Object {
            "cells": Array [
              Object {
                "title": "sit",
              },
            ],
          },
        ]
      }
      selectVariant="checkbox"
      summary="lorem ipsum summary"
      variant={null}
    >
      <TableBody />
    </Table>
  </GridItem>
</Grid>
`;

exports[`Table Component should allow variations in table layout: borders and table header removed 1`] = `
<Grid
  className="ins-inventory-list"
>
  <GridItem
    span={12}
  >
    <Table
      aria-label="t(curiosity-inventory.tableAriaLabel, {\\"appName\\":\\"Subscriptions\\"})"
      borders={false}
      canCollapseAll={false}
      canSelectAll={true}
      canSortFavorites={true}
      cells={
        Array [
          Object {
            "title": "lorem ipsum",
          },
        ]
      }
      className="curiosity-table-no-border "
      collapseAllAriaLabel=""
      contentId="expanded-content"
      dropdownDirection="down"
      dropdownPosition="right"
      expandId="expandable-toggle"
      gridBreakPoint="grid-md"
      isHeaderSelectDisabled={false}
      isNested={false}
      isStickyHeader={false}
      isTreeTable={false}
      ouiaSafe={true}
      role="grid"
      rowLabeledBy="simple-node"
      rows={
        Array [
          Object {
            "cells": Array [
              Object {
                "title": "dolor",
              },
            ],
          },
          Object {
            "cells": Array [
              Object {
                "title": "sit",
              },
            ],
          },
        ]
      }
      selectVariant="checkbox"
      summary="t(curiosity-inventory.tableSummary)"
      variant={null}
    >
      <TableBody />
    </Table>
  </GridItem>
</Grid>
`;

exports[`Table Component should allow variations in table layout: className and variant 1`] = `
<Grid
  className="ins-inventory-list"
>
  <GridItem
    span={12}
  >
    <Table
      aria-label="lorem ipsum aria-label"
      borders={false}
      canCollapseAll={false}
      canSelectAll={true}
      canSortFavorites={true}
      cells={
        Array [
          Object {
            "title": "lorem ipsum",
          },
        ]
      }
      className="curiosity-table-no-border lorem-ipsum-class"
      collapseAllAriaLabel=""
      contentId="expanded-content"
      dropdownDirection="down"
      dropdownPosition="right"
      expandId="expandable-toggle"
      gridBreakPoint="grid-md"
      isHeaderSelectDisabled={false}
      isNested={false}
      isStickyHeader={false}
      isTreeTable={false}
      ouiaSafe={true}
      role="grid"
      rowLabeledBy="simple-node"
      rows={
        Array [
          Object {
            "cells": Array [
              Object {
                "title": "dolor",
              },
            ],
          },
          Object {
            "cells": Array [
              Object {
                "title": "sit",
              },
            ],
          },
        ]
      }
      selectVariant="checkbox"
      summary="lorem ipsum summary"
      variant="compact"
    >
      <TableBody />
    </Table>
  </GridItem>
</Grid>
`;

exports[`Table Component should allow variations in table layout: generated rows 1`] = `
<Grid
  className="ins-inventory-list"
>
  <GridItem
    span={12}
  >
    <Table
      aria-label="t(curiosity-inventory.tableAriaLabel, {\\"appName\\":\\"Subscriptions\\"})"
      borders={true}
      canCollapseAll={false}
      canSelectAll={true}
      canSortFavorites={true}
      cells={
        Array [
          Object {
            "title": "lorem ipsum",
          },
        ]
      }
      className="curiosity-table "
      collapseAllAriaLabel=""
      contentId="expanded-content"
      dropdownDirection="down"
      dropdownPosition="right"
      expandId="expandable-toggle"
      gridBreakPoint="grid-md"
      isHeaderSelectDisabled={false}
      isNested={false}
      isStickyHeader={false}
      isTreeTable={false}
      ouiaSafe={true}
      role="grid"
      rowLabeledBy="simple-node"
      rows={
        Array [
          Object {
            "cells": Array [
              Object {
                "title": "dolor",
              },
            ],
          },
          Object {
            "cells": Array [
              Object {
                "title": "sit",
              },
            ],
          },
        ]
      }
      selectVariant="checkbox"
      summary="t(curiosity-inventory.tableSummary)"
      variant={null}
    >
      <TableHeader />
      <TableBody />
    </Table>
  </GridItem>
</Grid>
`;

exports[`Table Component should pass child components, nodes when there are no rows: children 1`] = `
<Grid
  className="ins-inventory-list"
>
  <GridItem
    span={12}
  >
    <Table
      aria-label="t(curiosity-inventory.tableAriaLabel, {\\"appName\\":\\"Subscriptions\\"})"
      borders={true}
      canCollapseAll={false}
      canSelectAll={true}
      canSortFavorites={true}
      cells={
        Array [
          Object {
            "title": "lorem ipsum",
          },
        ]
      }
      className="curiosity-table "
      collapseAllAriaLabel=""
      contentId="expanded-content"
      dropdownDirection="down"
      dropdownPosition="right"
      expandId="expandable-toggle"
      gridBreakPoint="grid-md"
      isHeaderSelectDisabled={false}
      isNested={false}
      isStickyHeader={false}
      isTreeTable={false}
      ouiaSafe={true}
      role="grid"
      rowLabeledBy="simple-node"
      rows={Array []}
      selectVariant="checkbox"
      summary="t(curiosity-inventory.tableSummary)"
      variant={null}
    >
      <TableHeader />
      <TableBody />
    </Table>
    Loading...
  </GridItem>
</Grid>
`;

exports[`Table Component should render a non-connected component: non-connected 1`] = `
<Grid
  className="ins-inventory-list"
>
  <GridItem
    span={12}
  >
    <Table
      aria-label="t(curiosity-inventory.tableAriaLabel, {\\"appName\\":\\"Subscriptions\\"})"
      borders={true}
      canCollapseAll={false}
      canSelectAll={true}
      canSortFavorites={true}
      cells={
        Array [
          Object {
            "title": "lorem",
          },
          Object {
            "title": "ipsum",
          },
          Object {
            "title": "dolor",
          },
          Object {
            "title": "sit",
          },
        ]
      }
      className="curiosity-table "
      collapseAllAriaLabel=""
      contentId="expanded-content"
      dropdownDirection="down"
      dropdownPosition="right"
      expandId="expandable-toggle"
      gridBreakPoint="grid-md"
      isHeaderSelectDisabled={false}
      isNested={false}
      isStickyHeader={false}
      isTreeTable={false}
      ouiaSafe={true}
      role="grid"
      rowLabeledBy="simple-node"
      rows={Array []}
      selectVariant="checkbox"
      summary="t(curiosity-inventory.tableSummary)"
      variant={null}
    >
      <TableHeader />
      <TableBody />
    </Table>
    <TableEmpty
      icon={[Function]}
      message="t(curiosity-inventory.tableEmptyInventoryMessage)"
      tableHeading="h2"
      title="t(curiosity-inventory.tableEmptyInventoryTitle)"
      variant="small"
    />
  </GridItem>
</Grid>
`;
